src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -55,29 +54,37 @@ internal abstract partial class AnalyzerDriver : IDisposable
         /// </summary>
         private readonly SeverityFilter _severityFilter;
 
-        // Lazy fields/properties
-        private CancellationTokenRegistration _queueRegistration;
         protected ImmutableArray<DiagnosticAnalyzer> Analyzers { get; }
         protected AnalyzerManager AnalyzerManager { get; }
+
+        // Lazy fields/properties
+        private Func<ISymbol, SyntaxReference, Compilation, CancellationToken, SyntaxNode> _lazyGetTopmostNodeForAnalysis;
```

Comment:
>_lazyGetTopmostNodeForAnalysis [](start = 91, length = 30)

Does this need to be an instance field, or lazily initialized?

Reply:
It is a field to avoid allocations as it is used in a hot path.

Reply:
> It is a field to avoid allocations as it is used in a hot path.

Can it be a `static readonly` field as it was before?

---
In reply to: [450430726](https://github.com/dotnet/roslyn/pull/45537#discussion_r450430726) [](ancestors = 450430726)

Reply:
I updated the implementation so now it depends on the `CompilationData` (instance field), which uses cached semantic model instead of creating a new one everytime.

Reply:
I see, thanks. In that case, consider removing the `CompilationData` parameter from `GetTopmostNodeForAnalysis()` and changing that method to an instance member instead since we already have one allocation per `AnalyzerDriver` for the delegate. That would avoid the closure class.

---
In reply to: [450433587](https://github.com/dotnet/roslyn/pull/45537#discussion_r450433587) [](ancestors = 450433587)

Reply:
Sounds good, thanks.